modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/repo/MetricRepository.java

```diff
@@ -17,12 +17,14 @@
 
 import com.expedia.adaptivealerting.modelservice.entity.Metric;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 /**
  * @author kashah
- *
  */
 public interface MetricRepository extends JpaRepository<Metric, Integer> {
 
-    Metric findMetricByMetricKey(String metricKey);
+    @Query("select m.id from Metric m where m.metricKey = :key")
+    Integer findIdByMetricKey(@Param("key") String key);
```

Comment:
Wouldn't we want to find the metric by the metric _hash_ here? Not the key. Not all metrics have a key.

Instead of finding the ID by hash, just find the whole Metric by hash using `findByHash`. This is more generally useful and doesn't change the query performance at all.

But we are not going to want to make two separate queries to get the models for a given metric--we can do it with a single query.

One constraint on the query is that we don't want metrics to know about models and vice versa. These are independent concepts. So what that means is that the `metric_model` link table should actually become its own entity in the Java representation. (Consider renaming the table to `metric_model_mapping` as well.) Let me suggest creating a `MetricModelMapping` entity along with a `MetricModelMappingRepo` repo. The `MetricModelMapping` should have a single reference to a `Metric` and another reference to a `Model`. Then create a JPA method on the repo called `findByMetricHash(String)`. This should automatically pull up all desired mappings in a single query. Moreover you will not need to define an explicit `@Query`.